processor/src/operations/decorators/mod.rs

```diff
@@ -7,9 +11,79 @@ impl Process {
     // DEBUGGING
     // --------------------------------------------------------------------------------------------
 
-    /// TODO: implement
-    pub fn op_debug(&self) -> Result<(), ExecutionError> {
-        unimplemented!();
+    /// Prints out debugging information based on options passed.
+    pub fn op_debug(&mut self, options: DebugOptions) -> Result<(), ExecutionError> {
+        match options {
+            DebugOptions::All => {
+                self.print_stack(None);
+                self.print_mem(None, None);
+            }
+            DebugOptions::Stack(n) => self.print_stack(n),
+            DebugOptions::Mem(n, m) => self.print_mem(n, m),
+            DebugOptions::Local(_) => self.print_local(),
+        }
+
+        Ok(())
+    }
+
+    // DEBUG HELPER
+    // ---------------------------------------------------------------------------------
+
+    /// Prints stack information for debugging.
+    /// If n is passed, this prints the top n items in the stack.
+    fn print_stack(&self, n: Option<usize>) {
+        let states = self.stack.get_values(n);
+        for (i, value) in states.iter().enumerate() {
+            info!("stack [ index: {:?}, value: {:?} ]", i, value.as_int());
+        }
```

Comment:
For the stack, I think it's OK if we don't print out the indexes. Just a comma separated list of values should be fine.

Also, I wonder if it would be good to add info about stack depth as well some how. I think it would be useful to know that the total depth is x but we printed out only y items.